ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/filter/FilterManager.java

```diff
@@ -162,35 +162,17 @@ public void recordBlockEvent(final BlockAddedEvent event, final Blockchain block
           }
         });
 
-    checkBlockchainForMatchingLogsForFilters();
+    filterRepository.getFiltersOfType(LogFilter.class).forEach(this::addNewMatchingLogs);
   }
 
-  private void checkBlockchainForMatchingLogsForFilters() {
-    filterRepository.getFiltersOfType(LogFilter.class).forEach(this::addMatchingLogs);
-  }
-
-  private void addMatchingLogs(final LogFilter filter) {
-    final long headBlockNumber = blockchainQueries.headBlockNumber();
+  private void addNewMatchingLogs(final LogFilter filter) {
```

Comment:
I renamed this method to clarify that we will only be looking for new logs here.  The `fromBlockNumber` is set to the head of the chain.